package/src/utilities/add-integration.ts

```diff
@@ -0,0 +1,62 @@
+import { type AstroIntegration, type HookParameters } from "astro";
+import { hasIntegration } from "./has-integration.js";
+
+export type addIntegrationUserOptionalParams = {
+	ensureUnique?: boolean;
+};
+
+export type addIntegrationUserParams = {
+	integration: AstroIntegration;
+	options?: addIntegrationUserOptionalParams
+}
+
+type AddIntegrationParams = addIntegrationUserParams & {
+	updateConfig: HookParameters<"astro:config:setup">["updateConfig"];
+	config: HookParameters<"astro:config:setup">["config"];
+	logger: HookParameters<"astro:config:setup">["logger"];
+};
```

Comment:
Can we just have one time similarly to my comment above? I think `integration` and `ensureUnique` should be at the top level. Then in the AIK plugin you can use `Pick`